redbpf-macros/src/lib.rs

```diff
@@ -168,6 +168,16 @@ pub fn impl_network_buffer_array(_: TokenStream) -> TokenStream {
 /// **NOTE:** The `#[map("foo")` (which uses link section `maps/foo`) has
 /// been deprecated in favor of `#[map]` or `#[map(link_section = "maps/foo")]`
 ///
+/// If you wish to share maps which already exist and are pinned to files, use
+/// `#[map(from_pin_file = "/path/to/pin/file")]`. This adds
+/// `mapxattr/maps/<item_name>` section to the resulting ELF binary. When the
+/// section is parsed by
+/// [redbpf::Module::parse](../../redbpf/struct.Module.html), redbpf makes BPF
+/// programs share the pinned map.
+///
+/// **CAUTION:** There should exist the designated pin files before the
```

Comment:
I'm not sure this is a great idea, as it's a bit of a catch 22. It might be best putting some fool-proof logic in there to make sure the error doesn't bubble up and cause hard to explain behaviour when trying to run a program.

Reply:
I agree with that this API is easy to fail. I will try figuring out how to enhance the API.

At first I considered to provide userspace API only, because it is intuitive and easy to find where the root cause is..

But it seemed hard to use that API and it seemed complicated to understand that unless new design of Module is introduced.. So I chose to offer attribute meta instead. It is easier to understand and simple to use.

I think that it would be better to split the Module::parse() into parse() and load(). And it allows users to do something between parse() and load(). But changing API like this would bring big difference to users and it really would bother them so I could not choose that. Maybe I can add other API without touching the original, but wouldn't it be better to do that when major version upgrade?